recipe/meta.yaml

```
@@ -1,208 +1,211 @@
+  - name: cudensitymat
+    version: 0.0.5
     build:
       number: {{ build_num }}
-      script:
-        - mkdir -p $PREFIX/include                                            # [linux]
-        - mv include/custatevec.h $PREFIX/include/                            # [linux]
-        - mkdir -p $PREFIX/lib                                                # [linux]
-        - mv lib/libcustatevec.so* $PREFIX/lib/                               # [linux]
+      string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ mpi_suffix }}"
+      script: "${RECIPE_DIR}/cudensitymat/cudm-install"
       run_exports:
-        - {{ pin_subpackage('custatevec', max_pin='x') }}
+        - {{ pin_subpackage("cudensitymat", max_pin="x") }} {{ mpi_prefix }}_*
       ignore_run_exports_from:
-        - {{ compiler('cuda') }}
-      missing_dso_whitelist:
-        - '*/libcublas.so*'    # [(cuda_compiler_version or "").startswith("11")]
-        - '*/libcublasLt.so*'  # [(cuda_compiler_version or "").startswith("11")]
-      post-link: post-link  # this is supported by conda-build, but undocumented
+        - {{ compiler("cuda") }}
+        - libcublas  # [cuda_compiler_version in ("12.0", "12.6")]
+      post-link: post-link
     requirements:
       build:
         - {{ stdlib('c') }}
-        - {{ compiler('c') }}
-        - {{ compiler('cxx') }}
-        - {{ compiler('cuda') }}
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - {{ compiler("cuda") }}
       host:
-        - libcublas  # [(cuda_compiler_version or "").startswith("12")]
-        - cuda-version {{ cuda_compiler_version }}
+        - cutensor >=2.0.2,<3
+        - cuda-version={{ cuda_compiler_version }}  # [cuda_compiler_version in ("11.8", "12.0", "12.6")]
+        - libcublas  # [cuda_compiler_version in ("12.0", "12.6")]
+        - cuda-cudart-dev # [cuda_compiler_version in ("12.0", "12.6")]
+        - {{ mpi }}  # [mpi != "nompi"]
       run:
-        - cudatoolkit                                     # [(cuda_compiler_version or "").startswith("11")]
-        - {{ pin_compatible('libcublas', max_pin='x') }}  # [(cuda_compiler_version or "").startswith("12")]
-        - cuda-version >={{ cuda_major }}.0,<{{ cuda_major+1 }}
-      # this should be set once conda-forge supports newer glibc
-      # https://github.com/conda-forge/conda-forge.github.io/issues/1941
-      # run_constrained:
-      #   - __glibc >=2.27  # [aarch64 or ppc64le]
-    # apparently conda-build supports full test section spec, the document is super outdated
+        - {{ pin_compatible("cutensor", max_pin="x") }}
+        - cuda-version >={{ cuda_major }}.0,<{{ cuda_major + 1 }}  # [cuda_compiler_version in ("11.8", "12.0", "12.6")]
+        - {{ pin_compatible("cuda-version", max_pin="x", min_pin="x") }} 
+        - libcublas  # [cuda_compiler_version in ("12.0", "12.6")]
+        - cudatoolkit  # [cuda_compiler_version == "11.8"]
+        - cuda-cudart-dev  # [cuda_compiler_version in ("12.0", "12.6")]
```

```suggestion
        - {{ pin_compatible("cutensor", max_pin="x") }}
        - {{ pin_compatible("cuda-version", max_pin="x", min_pin="x") }}  
        - libcublas # [cuda_compiler_version in ("12.0", "12.6")]
        - cudatoolkit  # [cuda_compiler_version == "11.8"]
        - cuda-cudart  # [cuda_compiler_version in ("12.0", "12.6")]
```
This dependency is redundant.

---

If i remember correctly, i added it since the build matrix is not generated without explicit dependency on cuda version (outside of `pin_compatible`).

---

Please check all output for duplicate dependency on cuda-version.

---

I tried it out again, and it indeed works without the duplicate dependency, thanks!

---

I believe that conda-smithy looks for `{{ compiler('cuda') }}` to determine if it should generate CUDA variants for a recipe.